orchagent/portsorch.cpp

```diff
@@ -1338,37 +1470,39 @@ bool PortsOrch::addVlanMember(Port vlan, Port port, string& tagging_mode)
     return true;
 }
 
-bool PortsOrch::removeVlanMember(Port vlan, Port port)
+bool PortsOrch::removeVlanMember(Port &vlan, Port &port)
 {
     SWSS_LOG_ENTER();
 
-    sai_status_t status = sai_vlan_api->remove_vlan_member(port.m_vlan_member_id);
+    sai_object_id_t vlan_member_id;
+    sai_vlan_tagging_mode_t sai_tagging_mode;
+    auto vlan_member = port.m_vlan_members.find(vlan.m_vlan_info.vlan_id);
+
+    /* Assert the port belongs to this VLAN */
+    assert (vlan_member != port.m_vlan_members.end());
+    sai_tagging_mode = vlan_member->second.vlan_mode;
+    vlan_member_id = vlan_member->second.vlan_member_id;
 
+    sai_status_t status = sai_vlan_api->remove_vlan_member(vlan_member_id);
     if (status != SAI_STATUS_SUCCESS)
     {
         SWSS_LOG_ERROR("Failed to remove member %s from VLAN %s vid:%hx vmid:%lx",
-                port.m_alias.c_str(), vlan.m_alias.c_str(), vlan.m_vlan_id, port.m_vlan_member_id);
+                port.m_alias.c_str(), vlan.m_alias.c_str(), vlan.m_vlan_info.vlan_id, vlan_member_id);
         return false;
     }
-
+    port.m_vlan_members.erase(vlan_member);
     SWSS_LOG_NOTICE("Remove member %s from VLAN %s lid:%hx vmid:%lx",
-            port.m_alias.c_str(), vlan.m_alias.c_str(), vlan.m_vlan_id, port.m_vlan_member_id);
-
-    sai_attribute_t attr;
-    attr.id = SAI_PORT_ATTR_PORT_VLAN_ID;
-    attr.value.u16 = DEFAULT_PORT_VLAN_ID;
+            port.m_alias.c_str(), vlan.m_alias.c_str(), vlan.m_vlan_info.vlan_id, vlan_member_id);
 
-    status = sai_port_api->set_port_attribute(port.m_port_id, &attr);
-    if (status != SAI_STATUS_SUCCESS)
+    /* Restore to default pvid if no pvid has been configured explictly and this port was in untagged mode */
+    if (port.m_port_vlan_id == DEFAULT_PORT_VLAN_ID && sai_tagging_mode == SAI_VLAN_TAGGING_MODE_UNTAGGED)
```

Comment:
>port.m_port_vlan_id == DEFAULT_PORT_VLAN_ID &&  [](start = 8, length = 47)

not sure if we need this.

Reply:
As explained previously, it was for supporting pvid configuration from admin interface.
removed.